x-pack/plugins/lists/server/services/exception_lists/delete_exception_list_item.ts

```diff
@@ -43,3 +50,12 @@ export const deleteExceptionListItem = async ({
     return exceptionListItem;
   }
 };
+
+export const deleteExceptionListItemById = async ({
+  id,
+  namespaceType,
+  savedObjectsClient,
+}: DeleteExceptionListItemByIdOptions): Promise<void> => {
+  const savedObjectType = getSavedObjectType({ namespaceType });
+  await savedObjectsClient.delete(savedObjectType, id);
+};
```

Comment:
The function above this works where you can call:

```ts
deleteExceptionListItem({ itemId: undefined, id, namespaceType, savedObjectsClient  });
```
Are you just trying to avoid the extra call to get the item if it does not exist?

Reply:
That, and also it avoids having to check for a `null` return value as an error case - `savedObjectsClient.delete` should either succeed or throw

Reply:
Yeah, I hear ya'.

I think there's been a bit of back and forth throughout the API's throughout the code with regards to throws and return statements that can contain errors. Sometimes people will want to do a throw and unwind the stack and then other times I see people wrapping those in a catch everywhere and then returning null and preferring the non-throw states (or wrapping it in an object/array that can contain an error).

In some cases people have been using more of a disjoint union such as the Either:
https://gcanti.github.io/fp-ts/modules/Either.ts.html

(Similar to RustLang but Rust has more first class citizen treatment of it)

So I have seen people kind of go back and forth depending on who I program with. You don't have to change anything, I'm fine with it, as it works for your use case. I think in newer contemporary  programming languages (golang, rustlang) most errors are now returned and not many throws are used unless they are something like a panic. Regardless of if they're using a disjointed union type, array type, etc...

TypeScript makes it a pain and close to impossible to strongly type throws or advertise what something can throw and when it can throw, so some circles of people tend to reduce their usage unless it's a panic type thing or they have nothing left to do way down the stack and kind of just say, "Ugh...Guess I should just unwind with a throw or I have to re-do all the return types throughout this piece of code".

I kind of do all of the above ;-) depending mostly on where I am in the code base and sometimes where my head is at, but for the validation type things which are right now returning arrays I know that we have been considering embracing more io-ts Either instead of flattening it out into an array and returning that as an Either at some point.

Again, no changes this is fine as is, just some thoughts of what I have noticed.